lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1952,7 +1955,7 @@ private function doMerge(
         $entity,
         array &$visited,
         $prevManagedCopy = null,
-        array $assoc = []
+        ?array $assoc = null
```

Comment:
When does `$assoc` get a `null` value injected? The second argument of the `updateAssociationWithMergedEntity()` method used inside `doMerge()` doesn't allow `null`.

Reply:
No args are passed here: https://github.com/doctrine/orm/blob/0f9cc194ae28881009db7c5a29b85c5abdef2f6d/lib/Doctrine/ORM/UnitOfWork.php#L1937

updateAssociationWithMergedEntity does not accept `null` but it doesn't accept an empty array either
https://github.com/doctrine/orm/blob/0f9cc194ae28881009db7c5a29b85c5abdef2f6d/lib/Doctrine/ORM/UnitOfWork.php#L2102

So I assume, that when it's called without args, the `updateAssociationWithMergedEntity` call is not done too.
IMHO it will be better to consider `null` as no value rather than `[]` as no value. And some null check can be added later for static analysis.

Reply:
It's a weird situation that shouldn't be handled in this PR.